scripts/wallet.js

```diff
@@ -329,16 +350,17 @@ export class Wallet {
      */
     loadAddresses() {
         if (this.isHD()) {
-            for (
-                let i = this.#loadedIndexes;
-                i <= this.#loadedIndexes + MAX_ACCOUNT_GAP;
-                i++
-            ) {
-                const path = this.getDerivationPath(0, i);
-                const address = this.#masterKey.getAddress(path);
-                this.#ownAddresses.set(address, path);
+            for (let c = 0; c < Wallet.chains; c++) {
```

Comment:
small optimization: instead of loading addresses for both chains, take the chain you want to load as input in `loadAddresses()`